File: 01-ub/src/bin/div_mul.rs
fn identity(x: i32, y: i32) -> i32 {
    x * y / y
}

fn print_identity(x: i32, y: i32) {
    println!("identity({}, {}) == {}", x, y, identity(x, y));
}

fn main() {
    print_identity(7, 5);
    print_identity(7, 0);
}

File: 01-ub/div_mul.cpp
#include <iostream>
#include <climits>

int identity(int x, int y)
{
    return x * y / y;
}

void print_identity(int x, int y)
{
    std::cout << "identity(" << x << ", " << y << ") == "
              << identity(x, y) << '\n';
}

int main()
{
    print_identity(7, 5);
}